### ISSUE TITLE
Login: Redesign the page

### ISSUE THREAD

**Issue by jasmussen** [WordPress/wporg-main-2022#241]
At the moment, login.wordpress.org looks like this:

![Before](https://user-images.githubusercontent.com/1204802/228235175-5b487955-857f-4f25-a2d2-b3c1c7bcc92b.png)

Let's refresh this. Here's a first pass:

<img width="960" alt="After" src="https://user-images.githubusercontent.com/1204802/228235208-273fb207-1550-4879-867e-485b2a122c97.png">

Let me know your feedback on the visuals!

**Comment by ryelle** [comment]
We can use this issue for tracking, but I'm pretty sure the code for login.wordpress.org lives in trac still: https://meta.trac.wordpress.org/browser/sites/trunk/wordpress.org/public_html/wp-content/themes/pub/wporg-login

**Comment by pablohoneyhoney** [comment]
I think the above is a styling improvement worth doing, pending further work on auth flows. Keep in mind the lost password as we implement it.

**Comment by bengreeley** [comment]
To add to Pablo's comment, there's also the new 2fa screens to consider. I'd imagine any changes to the `input` or `button` elements would be carried throughout the form, but it would be a good test scenario.

**Comment by jasmussen** [comment]
Took a stab at rolling out this refresh to the other sections. Sharing early here, as the full flow reveals a lot of additional opportunities for improvement, but it might be useful to see the work in progress to help inform that, so let me know any comments these mockups inspire. It might also be easier to see these designs in [the Figma file](https://www.figma.com/file/6KPZbXujMGDRWWDR5odjDE/WP.org-improvements?node-id=69%3A1437&t=0oyk2kijZwVc3sBz-1).

Reset password:

<img width="1120" alt="Reset password" src="https://user-images.githubusercontent.com/1204802/230902847-55a67f90-55a2-4ece-9596-d6f7234ecd00.png">

Register:

<img width="1120" alt="Create new account" src="https://user-images.githubusercontent.com/1204802/230902856-5e70a2a4-6a3c-4c0b-affe-f8345934c973.png">

Note that 2FA does not seem to have been implemented in the register flow quite yet, but I had a separate mockup for this that should be mostly compatible, give or take a few updates:

<img width="2120" alt="2fa" src="https://user-images.githubusercontent.com/1204802/230903147-11a2e0cd-2e1a-422e-92aa-532be1cde9b4.png">

A lot of opportunities reveal themselves in the full register flow:

* Revisiting the verbiage to be shorter, more precise
* Revisiting whether filling out your profile is worth doing during signup or not, this might be better filled out after the fact
* Exploring style and code reusage between this flow and the profile page, I imagine there's a lot of overlap, 2fa being one example
* Potentially making the registration flow into a "stepped" flow, not unlike how the new 2fa flow itself is built.

Nevertheless, share your notes!

**Comment by jasmussen** [comment]
Here's a very quick mockup of a reduced setup, that omits profile information and simplifies in a few other places:

<img width="3200" alt="Log in and register flow" src="https://user-images.githubusercontent.com/1204802/230904178-0b028ab4-4544-42d0-95ab-22b9847d2f11.png">

* Top align register pages, vs. being vertically centered for the main login and reset password pages.
* Remove profile information entirely, simplifying the password setting flow.
* Position the password notice as a replacement for the password field, improving context.

**Comment by fcoveram** [comment]
The new style looks nice. Clean and straightforward. I have some thoughts and ideas about the proposal.

## User account flows

> Revisiting whether filling out your profile is worth doing during signup or not, this might be better filled out after the fact

I am hesitant about covering this change now. Mapping all flows to revisit the process friction and possible copy inconsistencies sounds better as a first step for a flow redesign. I agree that filling a profile is unnecessary, but I am also curious about the different emails you get during these different flows.

So we can go with revisiting the flow or updating the style. Both efforts are valuable.

## Style update

Regarding the style update. I checked the site's Design Library and the other designs made, and I propose the following changes:
* Add a heading with `paragraph (large)` style on each page to reinforce the page's purpose. Especially in cases like "Link expired" or single-goal pages.
* Bring back the helper text in "Username" field. The string limitation prevents users from frustration if they don't know which characters are not allowed.
* Align WordPress logo to the left.
* Align CTA and secondary action (below CTA) in the center.
* Put the bottom links outside the content area as both actions are site-wise.
* Use the [Callout colors](https://www.figma.com/file/HOJTpCFfa3tR0EccUlu0CM/WordPress.org-Design-Library?node-id=1264-4116&t=ia6WZc14n7N25ToU-4) for the input alerts. This site-wise consistency is necessary.
* I tweaked some spacings based on the Design Library:
  * Move CTA and secondary buttons out of the form wrapper.
  * Reduce space to `30px` between sections in the content area (image attached below).

<img width="350" alt="CleanShot 2023-04-12 at 11 38 42@2x" src="https://user-images.githubusercontent.com/895819/231418987-c2406f74-6ea3-4dbf-a078-e8990b3b1345.png">

To not overload this message, here are attached some pages based on @jasmussen mockups, and here is the [whole flow](https://www.figma.com/file/6KPZbXujMGDRWWDR5odjDE/WP.org-improvements?node-id=95-1626&t=1Vh3pj47RiLTox8d-4).

| Page | Mockup |
|:---|:---|
| Register | ![image](https://user-images.githubusercontent.com/895819/231420196-b412e955-b78c-4154-8bf4-0d78d32a9180.png) |
| Register. Error fields | ![image](https://user-images.githubusercontent.com/895819/231420832-88d41ba1-8294-4621-8f93-de836476822e.png) |
| Profile. Password message | ![image](https://user-images.githubusercontent.com/895819/231420270-90934156-83fb-4df8-9eb1-714cd97de298.png) |

What do you think of this?

**Comment by jasmussen** [comment]
Nice work Francisco, and I like tackling this in two phases, the first of which is mainly visual. Profile or not can be a discussion for a different time.

It occurs to me that we probably should include iconography in the weak/strong indicator, as well as the invalid email/username notices, so as to accommodate folks with red/green color blindness. 

Small detail, in my mockup I rephrased the "Only lower case letters ..." help text to avoid the widow and keep it on one line.

**Comment by fcoveram** [comment]
> It occurs to me that we probably should include iconography in the weak/strong indicator, as well as the invalid email/username notices, so as to accommodate folks with red/green color blindness.

+1 to this idea. I tried different options, and I like the bottom one in the image below.

![image](https://user-images.githubusercontent.com/895819/231473321-99a4503a-16f1-4aec-b774-171e8ff0e099.png)

The helper text looks like an input validator when it's placed inside a box, like in [error](https://user-images.githubusercontent.com/895819/231420832-88d41ba1-8294-4621-8f93-de836476822e.png) and [success](https://user-images.githubusercontent.com/895819/231420270-90934156-83fb-4df8-9eb1-714cd97de298.png) examples. From my understanding, a weak password meets the field requirement, therefore, it makes sense to follow the helper text style of username.

> Small detail, in my mockup I rephrased the "Only lower case letters ..." help text to avoid the widow and keep it on one line.

I didn't get this. The screenshot in Figma file has the same copy as helper text below Username field.

**Comment by jasmussen** [comment]
Options look good, the helper text validator works well. If we do go with a background, I'd color code it. It seems okay to use the stoplight colors if we combine it with text/iconography.

But from these screenshots, the no-background options look great. I don't think we need colors, there, though, can just use the default gray helper text colors. Nice work!

**Comment by StevenDufresne** [comment]
@panchovm @jasmussen 

Is [version 3 in figma](https://www.figma.com/file/6KPZbXujMGDRWWDR5odjDE/WP.org-improvements?node-id=97-2146&t=M6GoCtQeJS1eJFQC-0) the version we are going with? Did we settle on the final password strength visuals?

**Comment by fcoveram** [comment]
Yes @StevenDufresne, we're going with [flow 3](https://www.figma.com/file/6KPZbXujMGDRWWDR5odjDE/WP.org-improvements?node-id=97-2146&t=oqk78TFbrRv9wQ5w-4). I updated the mockups with the final password helper design.
